app/scripts/controllers/transactions/lib/util.js

```diff
@@ -127,10 +128,21 @@ function ensureProperTransactionEnvelopeTypeProvided(txParams, field) {
  * Validates the given tx parameters
  *
  * @param {object} txParams - the tx params
+ * @param {string} type - type of the transaction
```

Comment:
I think we'll want to update `normalizeAndValidateTxParams` to support the new param as well